internal/terraform/transform_config.go

```diff
@@ -28,6 +28,9 @@ type ConfigTransformer struct {
 	// Mode will only add resources that match the given mode
 	ModeFilter bool
 	Mode       addrs.ResourceMode
+
+	// destroyPlan indicated this is being called from a destroy plan.
+	destroyPlan bool
```

Comment:
It feels a little weird to me that this flag essentially means to disable the behavior of this transformer entirely, but it's named as if it's going to modify the behavior in a more subtle way than that.

I guess being able to flag it off is probably easier than dynamically omitting it from the list of graph builders given that we currently just write those out as a slice constructor, but maybe it would read more clearly to just call this `disable`, `ignoreConfig` or similar instead, so that e.g. the caller will say `ignoreConfig: b.destroy` which (to me at least) reads as "ignore configuration if we are in destroy mode".